riscv/debug_module.cc

```diff
@@ -585,9 +587,7 @@ bool debug_module_t::perform_abstract_command()
     unsigned i = 0;
     if (get_field(command, AC_ACCESS_REGISTER_TRANSFER)) {
 
-      if (regno < 0x1000 && progbufsize < 2) {
-        // Make the debugger use the program buffer if it's available, so it
-        // can test both use cases.
+      if (regno < 0x1000 && support_abstract_csr_access) {
```

Comment:
Does this need to be `(support_abstract_csr_access || progbufsize < 2)`?  Or perhaps earlier, `assert(support_abstract_csr_access || progbufsize >= 2);`

Reply:
No, this test is correct. It's testing that the value is <0x1000 (the CSRs have values from 0-0xfff), and that abstract CSR access is enabled.